common/src/main/java/org/conscrypt/Conscrypt.java

```diff
@@ -67,6 +67,34 @@ public static X509TrustManager getDefaultX509TrustManager() throws KeyManagement
     }
 
     /**
+     * Utility that exposes common TLS constants.
+     */
+    public static final class Constants {
+        private Constants() {}
+
+        /**
+         * Returns the maximum length (in bytes) of plaintext data.
+         */
+        public static int maxPlaintextLength() {
+            return NativeConstants.SSL3_RT_MAX_PLAIN_LENGTH;
+        }
+
+        /**
+         * Returns the maximum length (in bytes) of an encrypted packet.
+         */
+        public static int maxEncryptedPacketLength() {
+            return NativeConstants.SSL3_RT_MAX_PACKET_SIZE;
+        }
+
+        /**
+         * Returns the length (in bytes) of an encrypted packet header.
```

Comment:
I'm confused by the mention of "encrypted" here and suspect this may not exposing what you're trying to expose.

The record-level header in TLS is the same between encrypted and unencrypted records. It's 5 bytes. (Currently! there is nothing, save for buggy middleboxes, preventing future iterations of the protocol from changing the encrypted header. How stable is this API considered to be? If stable, the bar is, as always, very high for adding anything. If unstable, this needs to be documented to signal that zero cares need be given on our end about forcing you to break it.)

This is *not* the same as the encryption overhead of a record and is *not* the same as the amount of prefix added to a record. Both of those depend on the version and cipher suite. Sometimes, encryption adds a 5 byte prefix and a variable-length suffix. Sometimes it adds a 5 byte prefix and a fixed-length suffix. Sometimes, it adds a larger fixed-byte prefix and variable-length suffix. Sometimes it adds a larger fixed-byte prefix and a fixed-length suffix.

What are you actually trying to do here?